emission/ce186/utility_model.py

```diff
@@ -0,0 +1,505 @@
+        to_return.append(area)
+    return to_return
+
+def get_time_of_trip(trip):
+    return (trip.end_time - trip.start_time).seconds
+
+def get_normalized_times(lst_of_trips):
+    counter = emmc.Counter()
+    i = 0
+    for trip in lst_of_trips:
+        counter[i] = get_time_of_trip(trip)
+    counter.normalize()
+    to_return = []
+    for i in range(len(lst_of_trips)):
+        to_return.append(counter[i])
+    return to_return
+
+def get_sweat_factor(trip):
+    start = trip.trip_start_location.to_tuple()
+    end = trip.trip_end_location.to_tuple()
+    chng = get_elevation_change(start, end)
+    if chng[0] < chng[1]:
+        return 0
+    return 71.112*chng[0] + 148.09
+
+def get_normalized_sweat(lst_of_trips):
+    counter = emmc.Counter()
+    i = 0
+    for trip in lst_of_trips:
+        counter[i] = get_sweat_factor(trip)
+    counter.normalize()
+    to_return = []
+    for i in range(len(lst_of_trips)):
+        to_return.append(counter[i])
+    return to_return
+
+class Area:
+
+    """ Area class """ 
+
+    def __init__(self, name, tl, br, beauty=None, time_to_noise=None):
+        self.name = name
+        self.bounding_box = (tl, br)
+        self.beauty = beauty
+        self.time_to_noise = time_to_noise
+        self.times = set()
+
+    def point_in_area(self, lat, lng):
+        return in_bounding_box(lat, lng, self.bounding_box)
+
+    def add_time(self, time):
+        self.times.add(time)
+
+    def get_crowd(self):
+        return len(self.times)
+
+    def update_times(self, time_by):
+        for time in self.times:
+            if time < time_by:
+                self.times.remove(time)
+
+    def update_to_now(self):
+        self.update_times(datetime.datetime.now())
+
+    def normalize_sounds(self):
+        counter = emmc.Counter()
+        for k,v in self.time_to_noise.iteritems():
+            counter[k] = v
+        counter.normalize()
+        self.time_to_noise = counter
+
+
+def in_bounding_box(lat, lon, bounding_box):
+    return bounding_box[1][0] <= lat and lat <= bounding_box[0][0] and bounding_box[0][1] <= lon and lon <= bounding_box[1][1]
+    
+def parse_noise():
+    noise_file = open("emission/ce186/noise_data.csv")
+    sproul_noises, glade_noises, wellmen_noises, leconte_noises = {}, {}, {}, {}
+    sproul_tl, sproul_br = (37.870637,-122.259722), (37.868926,-122.259005)
+    glade_tl, glade_br = (37.87359,-122.260098), (37.872707,-122.258687)
+    wellmen_tl, wellmen_br = (37.873045,-122.263377), (37.872501,-122.261803)
+    lecont_tl, lecont_br = (37.873278,-122.256959), (37.872277,-122.25639)
+    time = datetime.datetime(2040, 10, 10, 6, 0, 0)
+    td = datetime.timedelta(minutes=10)
+    for l in noise_file:
+        l = l.split(',')
+        sproul_noises[time] = float(l[0])
+        glade_noises[time] = float(l[1])
+        wellmen_noises[time] = float(l[2])
+        leconte_noises[time] = float(l[3])
+    sproul = Area("sproul", sproul_tl, sproul_br, time_to_noise=sproul_noises)
```

Comment:
Future: You could put the area configurations into mongodb and retrieve them by using a geojson query instead of doing a manual bbox. This is fine if you have 5 areas, but doesn't scale.